src/task_manager.cc

```diff
@@ -43,13 +43,17 @@ namespace detail {
 	}
 
 	void task_manager::notify_horizon_executed(task_id tid) {
+#ifndef NDEBUG
+		std::lock_guard lock{task_mutex};
```

Comment:
What is the intended scope of this `lock_guard`? The way it is now it will remain locked for the entire function. I assume that is not intended, and it should only be locked within the `#ifndef` block, which would require an additional block scope.